docs/contracts/staking-router.md

```diff
@@ -3,9 +3,843 @@
+
+Because Lido does not account for validator performance, the protocol fee is distributed between modules proportionally to active validators and the specified module fee. For example, a module with 75% of all validators in the protocol and a 5% module fee will receive 3.75% of the total rewards across the protocol. This means that if the modules' fee and treasury fee do not exceed 10%, the total protocol fee will not either, no matter how many modules there are. There is also an edge case where the module is stopped for emergency while its validators are still active. In this case the module fee will be transferred to the treasury and once the module is back online, the rewards will be returned back to the module from the treasury.
+
+The distribution function itself works as follows:
+
+1. The function first loads the current state of the staking modules into a cache (`_loadStakingModulesCache`) and calculates the number of these modules (`stakingModulesCount`).
+    
+2. If there are no staking modules or no active validators in the system, it returns an empty response.
+    
+3. Otherwise, it initializes arrays to store the module IDs (`stakingModuleIds`), the addresses of reward recipients (`recipients`), and the fees of each recipient (`stakingModuleFees`). It also sets the `precisionPoints` to a constant `FEE_PRECISION_POINTS`, which represents the base precision number that constitutes 100% fee.
+    
+4. Then it loops through each staking module. For each module that has at least one active validator, it:
+    
+    - Stores the module ID and recipient address in the respective arrays.
+    - Calculates the `stakingModuleValidatorsShare`, which is the proportion of total active validators that are part of this staking module.
+    - Calculates the `stakingModuleFee` as the product of `stakingModuleValidatorsShare` and the fee of the staking module divided by `TOTAL_BASIS_POINTS` (i.e., the proportion of the staking module's fee to the total possible fees). If the module is not stopped, this fee is stored in the `stakingModuleFees` array.
+    - Adds to `totalFee` the sum of the staking module's fee and a fee going to the treasury (calculated similarly to `stakingModuleFee`), where the treasury is a central pool of funds.
+5. After looping through all modules, it makes an assertion that `totalFee` doesn't exceed 100% (represented by `precisionPoints`).
+    
+6. If there are staking modules with no active validators, it shrinks the `stakingModuleIds`, `recipients`, and `stakingModuleFees` arrays to exclude those modules.
+    
+Finally, the function returns five arrays/values: `recipients`, `stakingModuleIds`, `stakingModuleFees`, `totalFee`, and `precisionPoints`. These give the caller an overview of how rewards are distributed amongst the staking modules.
+
+## View methods
+
+### `getLido`
+
+Returns the address of the core `Lido` contract.
+
+```solidity
+function getLido() public view returns (address)
+```
+
+**Returns:**
```

Comment:
Maybe exclude this at all?

Reply:
Do you mean the Returns section or the function itself?

Reply:
the section